lib_eio_posix/eio_posix_stubs.c

```diff
@@ -185,3 +187,17 @@ CAMLprim value caml_eio_posix_renameat(value v_old_fd, value v_old_path, value v
   if (ret == -1) uerror("renameat", v_old_path);
   CAMLreturn(Val_unit);
 }
+
+CAMLprim value caml_eio_posix_spawn(value v_errors, value v_actions) {
+  CAMLparam1(v_actions);
+  pid_t child_pid;
+
+  child_pid = fork();
+  if (child_pid == 0) {
+    eio_unix_run_fork_actions(Int_val(v_errors), v_actions);
+  } else if (child_pid < 0) {
```

Comment:
```suggestion
  } else if (child_pid == -1) {
```

Reply:
Exactly, PIDs can't be negative and fork cannot return anything below -1.

Reply:
If it never returns anything below -1 then this code change makes no difference. And if it does, then the original version was safer.

Reply:
it was just more "precise" :smile: